polygeist/lib/polygeist/Passes/ConvertToLLVMABI.cpp

```diff
@@ -0,0 +1,123 @@
+//===- ConvertToLLVMABI.cpp - Convert to LLVM Pointer ABI -----------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This pass currently only handles SYCL Kernel functions.
+// This pass converts all arguments with MemRef type to LLVM pointer type,
+// and replace all uses of the original argument with a
+// `polygeist.pointer2memref` of the new argument.
+//
+//===----------------------------------------------------------------------===//
+
+#include "PassDetails.h"
+#include "mlir/Conversion/LLVMCommon/Pattern.h"
+#include "mlir/Conversion/LLVMCommon/TypeConverter.h"
+#include "mlir/Dialect/GPU/IR/GPUDialect.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/Dialect/MemRef/IR/MemRef.h"
+#include "mlir/Transforms/DialectConversion.h"
+#include "llvm/Support/Debug.h"
+
+#define DEBUG_TYPE "convert-to-llvm-abi"
+
+using namespace mlir;
+
+namespace {
+
+class GPUFuncLowering : public OpRewritePattern<gpu::GPUFuncOp> {
```

Comment:
Eventually we will need to handle `func.func` (or the LLVM dialect func) as well.

Reply:
agree, only handling `gpu.func` with kernel attribute is a short-term limitation in this PR.